src/Core/Services/AuthService.cs

```diff
@@ -694,5 +699,22 @@ private async Task<PasswordlessLoginResponse> PopulateFingerprintPhraseAsync(Pas
             passwordlessLogin.FingerprintPhrase = string.Join("-", await _cryptoService.GetFingerprintAsync(userEmail, CoreHelpers.Base64UrlDecode(passwordlessLogin.PublicKey)));
             return passwordlessLogin;
         }
+
+        public async Task CreateNewSsoUserAsync(string organizationSsoId)
+        {
+            var orgAutoEnrollStatusResponse = await _apiService.GetOrganizationAutoEnrollStatusAsync(organizationSsoId);
+            var randomBytes = _cryptoFunctionService.RandomBytes(64);
+            var userKey = new UserKey(randomBytes);
+            var keyPair = await _cryptoService.MakeKeyPairAsync(userKey);
```

Comment:
⛏️ In order to avoid having `Item1` and `Item2`, deconstruct the tuple to have more meaningful names.